### DISCUSSION TITLE
Language converter: English to Welsh

### DISCUSSION THREAD

**Discussion by Wesv1** [community/community#121574] [Programming Help]
### Select Topic Area

Question

### Body

Hello 

New to Github 

Was wandering if someone can help me

I'm looking to convert English text to Welsh text, and for that text, to then be spoken 

I've used Gtts, pytts3x, DeepL and Googletrans, but nothing seems to be working

Can anyone provide some insight if you know a way to do it please?

Thanks

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @Wesv1 , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Comment by YassineCherkaoui** [comment]
Hello! 
It sounds like you're working on a text-to-speech project to convert English text to Welsh text and then have it spoken. While GTTS, pyttsx3, DeepL, and Googletrans are useful tools, they might not directly support Welsh text. 

Have you tried looking for Welsh text-to-speech libraries specifically? There might be libraries or APIs that support Welsh language processing.

Additionally, you could explore text translation APIs that support Welsh. Google Cloud Translation API, for example, supports Welsh translation. You could translate your English text to Welsh using this API before passing it to a text-to-speech library.

If you're still encountering issues, feel free to share more details about the specific problems you're facing, and I'll do my best help

**Reply by ebndev** [reply]
Hey @Wesv1! Did you get a chance to look at @YassineCherkaoui's suggestions?

Let us know if you've made any progress, it sounds like a great project :)

**Reply by Wesv1** [reply]
Hi.  

Not had a chance yet due to work and university, however, will definitely be trying in the next couple of weeks. Thank you all

**Reply by GeronimoKTT** [reply]
hey @Wesv1 I strongly suggest [www.nemesyslabs.com](https://www.nemesyslabs.com/)

**Comment by techpond31** [comment]
👋Hello! 

To accomplish your goal, you can use libraries like `googletrans` and `pyttsx3` in conjunction. Here's a Python script to translate English text to Welsh and speak it using `pyttsx3`.

First, install the required packages:

```
pip install googletrans==4.0.0-rc1
pip install pyttsx3
```

Then, create a python script `text_translator.py`:

```python
from googletrans import Translator
import pyttsx3

# Function to translate text
def translate_text(text, dest='cy'):
    translator = Translator()
    translated = translator.translate(text, src='en', dest=dest)
    return translated.text

# Function to speak text
def speak_text(text):
    engine = pyttsx3.init()
    engine.say(text)
    engine.runAndWait()

# Get text from user
input_text = input("Enter text to translate and speak: ")

# Convert English text to Welsh text
translated_text = translate_text(input_text)

# Speak the Welsh text
speak_text(translated_text)
```

Execute the script by running:

```
python text_translator.py
```

This script creates a simple interface to translate English text to Welsh and speaks it using `pyttsx3`. Note that `googletrans` v4.0.0-rc1 is used here, as recent versions may not support translation to Welsh ('cy').

I hope this helps! Have fun experimenting with the script :)
